lib/battle/IBattleState.h

```diff
@@ -87,5 +87,6 @@ class DLL_LINKAGE IBattleState : public IBattleInfo
 	virtual void setWallState(int partOfWall, si8 state) = 0;
 
 	virtual void addObstacle(const ObstacleChanges & changes) = 0;
+	virtual void updateObstacle(const ObstacleChanges& changes) = 0;
```

Comment:
- This change requires `BattleStateMock` to be updated.
- Put spaces around `&`

Reply:
Updated.